test/language/statements/class/subclass/derived-class-return-override-for-of-arrow.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2021 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-ecmascript-function-objects-construct-argumentslist-newtarget
+description: >
+  ReferenceError when returning from a derived class constructor without calling
+  `super()` is thrown after the function body has been left, so an iterator
+  return handler can still call `super()`.
+---*/
+
+var iter = {
+  [Symbol.iterator]() {
+    return this;
+  },
+  next() {
+    return {done: false};
+  },
+  return() {
+    // Calls |super()|.
+    this.f();
+
+    return {done: true};
+  },
+};
+
+class C extends class {} {
+  constructor() {
+    iter.f = () => super();
+
+    for (var k of iter) {
+      return;
```

Comment:
this test is great! 

I'm just not sure if this should leave any note (comment/metadata) saying this will provoke the iterator's return.

Reply:
> this test is great!

Thanks! :-)

The description in the YAML block mentions that the "iterator return handler" is called. But I can also add a line comment here to clarify which code triggers the handler. WDYT?

Reply:
it should be ok without changes. I just had a random and noisy thought.